File: Python_Ruby/stock_portfolio/portfolio.py
# Will MacLean
# CSPP 51060
# portfolio.py

filename = "portfolio.txt"
total = 0
shares = 0
price = 0

print
print "Name \tShares \t Price"
print "---- \t------ \t -----"

file = open( filename, "r" )
for line in file:
	fields = line.split()
	shares = int(fields[1])
	price = float(fields[2])
	# print fields[0], "\t", shares, "\t", price
	print "%s \t%5.0f \t%6.2f" % (fields[0], shares, price) 
	total = total + shares * price

print "\nTotal cost: ", total, "\n"



	

File: Python_Ruby/stock_portfolio/portfolio.pl
# Will MacLean
# CSPP 51060
# portfolio.pl

$filename = "portfolio.txt";
$total = 0;
$shares = 0;
$price = 0;

print "\n";
print "Name \tShares \t Price\n";
print "---- \t------ \t -----\n";

open file, $filename or die $!;
while ($record = <file>) {
	# print $record;
	@fields = split(/ /, $record, 3);
	printf "%s \t%5.0f \t%6.2f \n", @fields[0], @fields[1], @fields[2];
	$total = $total + @fields[1] * @fields[2];
}

print "\nTotal cost: ", $total, "\n\n";
close(file);
